crates/red_knot_python_semantic/resources/mdtest/doc/public_type_undeclared_symbols.md

```diff
@@ -0,0 +1,115 @@
+# Public type of undeclared symbols
+
+## Summary
+
+One major deviation from the behavior of existing Python type checkers is our handling of 'public'
+types for undeclared symbols. This is best illustrated with an example:
+
+```py
+class Wrapper:
+    value = None
+
+wrapper = Wrapper()
+
+reveal_type(wrapper.value)  # revealed: Unknown | None
+
+wrapper.value = 1
+```
+
+Mypy and Pyright both infer a type of `None` for the type of `wrapper.value`. Consequently, both
+tools raise an error when trying to assign `1` to `wrapper.value`. But there is nothing wrong with
+this program. Raising an error here violates the [gradual guarantee] which states that *"Removing
+type annotations (making the program more dynamic) should not result in additional static type
+errors."*: If `value` were annotated with `int | None` here, Mypy and Pyright would not raise any
+errors.
```

Comment:
nit: "raising an error" to me makes it sound to me like we'll panic; I prefer to talk about "emitting an error"/"issuing a diagnostic".

(This might just be because I'm used to the Python context where there's a `raise` keyword, though

)
```suggestion
Mypy and Pyright both infer a type of `None` for the type of `wrapper.value`. Consequently, both
tools emit an error when trying to assign `1` to `wrapper.value`. But there is nothing wrong with
this program. Emitting an error here violates the [gradual guarantee] which states that *"Removing
type annotations (making the program more dynamic) should not result in additional static type
errors."*: If `value` were annotated with `int | None` here, Mypy and Pyright would not emit any
errors.
```